persistent-sqlite/Database/Persist/Sqlite.hs

```diff
@@ -646,27 +660,42 @@ sqlColumn noRef (Column name isNull typ def gen _cn _maxLen ref) = T.concat
     , mayGenerated gen
     , case ref of
         Nothing -> ""
+        -- This foreign key constraint is only legitimate if it is a reference
+        -- within the same schema.
         Just ColumnReference {crTableName=table, crFieldCascade=cascadeOpts} ->
-          if noRef then "" else " REFERENCES " <> escapeE table
+            if noRef
+                then ""
+                -- This foreign key constraint is only legitimate if it is a reference
+                -- within the same schema. It's a syntax error in SQLite to use a
+                -- dot-qualified name here, so we just escape the table name.
+                else " REFERENCES " <> escapeE table
             <> onDelete cascadeOpts <> onUpdate cascadeOpts
     ]
   where
     onDelete opts = maybe "" (T.append " ON DELETE " . renderCascadeAction) (fcOnDelete opts)
     onUpdate opts = maybe "" (T.append " ON UPDATE " . renderCascadeAction) (fcOnUpdate opts)
 
-sqlForeign :: ForeignDef -> Text
-sqlForeign fdef = T.concat $
+sqlForeign :: Maybe SchemaNameDB -> ForeignDef -> Text
+sqlForeign entitySchema fdef = T.concat $
     [ ", CONSTRAINT "
     , escapeC $ foreignConstraintNameDBName fdef
     , " FOREIGN KEY("
     , T.intercalate "," $ map (escapeF . snd. fst) $ foreignFields fdef
     , ") REFERENCES "
-    , escapeE $ foreignRefTableDBName fdef
+    , -- It's not possible for SQLite to maintain foreign key
+      -- constraints across databases (which Persistent calls "schemas").
+      -- In fact, it's a syntax error to use a schema qualifier in this
+      -- part of the SQL expression. So we omit the schema here, and throw
+      -- an error if this is a cross-schema reference.
```

Comment:
Yeah this bit feels a bit weird to me. Does sqlite actually support what we've referred to elsewhere as "schemas" (i.e. namespaces within the same database)? (Looks like [no](https://stackoverflow.com/a/33962375)?) Or are we grasping at some other sqlite functionality that we imagine could be made analogous to schemas? What happens if sqlite adds support for true schemas later?

Should sqlite maybe fall into the same category as MongoDB, where the actual backend just doesn't support them?